src/fosslight_scanner/_run_compare.py

```diff
@@ -0,0 +1,165 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# FOSSLight Scanner
+# Copyright (c) 2022 LG Electronics Inc.
+# SPDX-License-Identifier: Apache-2.0
+import os
+import sys
+import json
+import yaml
+import logging
+import xlsxwriter
+import codecs
+from pathlib import Path
+from bs4 import BeautifulSoup
+import fosslight_util.constant as constant
+from fosslight_util.compare_yaml import compare_yaml
+
+logger = logging.getLogger(constant.LOGGER_NAME)
+ADD = "add"
+DELETE = "delete"
+CHANGE = "change"
+
+
+def write_result_json_yaml(output_file, compared_result, file_ext):
+    ret = True
+    try:
+        with open(output_file, 'w') as f:
+            if file_ext == '.json':
+                json.dump(compared_result, f, indent=4, sort_keys=True)
+            elif file_ext == '.yaml':
+                yaml.dump(compared_result, f, sort_keys=True)
+    except Exception:
+        ret = False
+    return ret
+
+
+def parse_result_for_table(oi, status):
+    compared_row = []
+    if status == ADD:
+        oss_after = f"{oi['name']} ({oi['version']})"
+        license_after = f"{', '.join(oi['license'])}"
+        compared_row = [status, '', '', oss_after, license_after]
+    elif status == DELETE:
+        oss_before = f"{oi['name']} ({oi['version']})"
+        license_before = f"{', '.join(oi['license'])}"
+        compared_row = [status, oss_before, license_before, '', '']
+    elif status == CHANGE:
+        oss_before, oss_after, license_before, license_after = [], [], [], []
+        for prev_i in oi['prev']:
+            oss_before.append(f"{oi['name']} ({prev_i['version']})")
+            license_before.append(f"{', '.join(prev_i['license'])}")
+        for now_i in oi['now']:
+            oss_after.append(f"{oi['name']} ({now_i['version']})")
+            license_after.append(f"{', '.join(now_i['license'])}")
+        compared_row = [status, ' / '.join(oss_before), ' / '.join(license_before),
+                                ' / '.join(oss_after), ' / '.join(license_after)]
+    else:
+        logger.error(f"Not supported compared status: {status}")
+
+    return compared_row
+
+
+def get_sample_html():
+    RESOURCES_DIR = 'resources'
+    SAMPLE_HTML = 'bom_compare.html'
+    html_file = os.path.join(RESOURCES_DIR, SAMPLE_HTML)
+
+    try:
+        base_dir = sys._MEIPASS
+    except Exception:
+        base_dir = os.path.dirname(__file__)
+
+    file_withpath = os.path.join(base_dir, html_file)
+    try:
+        html_f = codecs.open(file_withpath, 'r', 'utf-8')
```

Comment:
If there is an error here, it seems that html_f cannot be returned.
`html_f ` with default value should be declared at the top like `html_file`.

Reply:
I changed it.